pkg/status/binding.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+
+	"github.com/kubestellar/kubestellar/pkg/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (c *Controller) syncBinding(ctx context.Context, key string) error {
+
+	isDeleted := false
+	binding, err := c.wdsKsClient.ControlV1alpha1().Bindings().Get(ctx, key, metav1.GetOptions{})
```

Comment:
This block can be ejected. The source of events and truth is the broker.
If a binding key gets here and no resolution exists, then `isDeleted` can be set to true.